packages/node/src/methods/state-channel/add.ts

```diff
@@ -2,18 +2,20 @@ import { StateChannel } from "@counterfactual/machine";
 import { AssetType } from "@counterfactual/types";
 import { bigNumberify } from "ethers/utils";
 
-import { NodeMessage } from "../../node";
 import { RequestHandler } from "../../request-handler";
+import { CreateMultisigMessage } from "../../types";
 
 /**
  * This creates an entry for an already-created multisig sent by a peer.
  * @param nodeMsg
  */
-export async function addMultisig(this: RequestHandler, nodeMsg: NodeMessage) {
-  const params = nodeMsg.data;
+export async function addMultisig(
+  this: RequestHandler,
+  nodeMsg: CreateMultisigMessage
+) {
   let stateChannel = new StateChannel(
-    params.multisigAddress,
-    params.owners
+    nodeMsg.data.multisigAddress,
+    nodeMsg.data.params.owners
```

Comment:
It seems to me like the notion of a "Node Message" / `nodeMsg` / `NodeMessage` should not be something that the internal implementation of the `Node` (i.e., the "methods") should know about. Perhaps it makes sense that it is only ever _received_ by the `RequestHandler` and then _passed_ to a "controller" and at that point it is dissolved into arguments passed into the actual function implementations.

I think we mostly have this right now, but maybe a way to clean up the code a bit would be to have a single location where `NodeMessages` are converted into function arguments with no other logic involved. For example:

```typescript
function reducer(requestHandler: RequestHandler, msg: NodeMessage) {
    switch (msg.type) {
        case NodeMessage.CREATE_MULTISIG:
            addMultisig({
                multisigAddress: msg.data.multisigAddress,
                owners: msg.data.params.owners,
            }, {
                networkContext: requestHandler.networkContext
            });
            break;
        ...
    }
}
```

This way we would benefit from:
- Easier unit testing for all of the "message handlers"
- Clean separation of `NodeMessage` and the actual functional implementation of the `Node`

Reply:
Sgtm